rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,302 @@
+
+One key difference is that the trainsheets do not record the destination for trips, so added trips only include one stop. RTR should assume a default pattern for these trips.
+
+## Long-term storage / querying
+For future use, events will be archived to S3, using Kinesis Firehose. They can either be sent to LAMP's folder, or a new folder.
+
+## Examples
+### Train departing 15 minutes later than scheduled
+- Inspector Alice (badge number: 123) is working at Mattapan
+- The 1:30am trip is being held 15 minutes to wait for a final Red Line train
+```json
+{
+  "specversion": "1.0",
+  "type": "com.mbta.ctd.glides.trip-updated.v1",
+  "source": "glides.mbta.com",
+  "id": "19fdb184-7dd6-4664-8472-04bd6177ec44",
+  "time": "2023-01-23T01:25:00-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "tripKey": {
+      "serviceDate": "2023-01-22",
+      "stop": "place-matt",
+      "time": "25:30:00"
+    },
+    "departureTime": "25:45:00"
+  }
+}
+```
+
+Note that the date of the the event is 2023-01-23, but the service date is 2023-01-22. The hour in the time itself is greater than 24 hours, to reflect that the time is the next calendar day.
+### Splitting a 2-car train into two 1-car trips
+- Inspector Alice (badge number: 123) is working at Boston College
+- There are two scheduled trips: 9:55am and 10:00am
+- Operator A (badge number: 456) and Operator B (badge number: 567) are scheduled to perform the 9:55am
+- Only 1 trainset (3800 and 3850) is available
+- In order to maintain headways, Inspector Alice will split the trainset
+- Operator A will take 3800 and run the 9:55am
+- Operator B will take 3850 and run the 10:00am
+
+```json
+{
+  "specversion": "1.0",
+  "type": "com.mbta.ctd.glides.operator-signed-in.v1",
+  "source": "glides.mbta.com",
+  "id": "19fdb184-7dd6-4664-8472-04bd6177ec44",
+  "time": "2023-01-20T09:45:10-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "operator": {
+      "badgeNumber": "456"
+    },
+    "signInAt": "2023-01-20T09:45:00-05:00"
+  }
+}
+# further specversion, source, time, and ID fields elided from the top level
+# actor elided from underneath "data"
+{
+  "type": "com.mbta.ctd.glides.operator-signed-in.v1",
+  "data": {
+    "operator": {
+      "badgeNumber": "567"
+    },
+    "signInAt": "2023-01-20T09:45:00-05:00"
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trip-dropped.v1",
+  "data": {
+    "tripKey": {
+      "serviceDate": "2022-01-20",
+      "stop": "place-lake",
+      "scheduledTime": "10:00:00"
+    },
+    "reason": "ran as single"
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trip-updated.v1",
+  "data": {
+    "tripKey": {
+      "serviceDate": "2022-01-20",
+      "stop": "place-lake",
+      "scheduledTime": "09:55:00"
+    },
+    "operators": [
+      {"badgeNumber": "456"}
+    ],
+    "consist": [
+      {"carNumber": "3800"}
+    ]
```

Comment:
Currently, there doesn't seem to be a way to convey "the trailer has been removed from this trip entirely" distinctly from "the trailer car number hasn't been entered yet". Should there be?

Reply:
@skyqrose is correct: that's why the value is `Optional X` to distinguish those two cases. This is also called out in the Guide-level explanation for this event.